src/topology/order/lower_topology.lean

```diff
@@ -0,0 +1,207 @@
+@[pattern] def to_lower : α ≃ with_lower_topology α := equiv.refl _
+
+/-- `of_lower` is the identity function from the `with_lower_topology` of a type.  -/
+@[pattern] def of_lower : with_lower_topology α ≃ α := equiv.refl _
+
+@[simp] lemma to_with_lower_topology_symm_eq : (@to_lower α).symm = of_lower := rfl
+@[simp] lemma of_with_lower_topology_symm_eq : (@of_lower α).symm = to_lower := rfl
+@[simp] lemma to_lower_of_lower (a : with_lower_topology α) : to_lower (of_lower a) = a := rfl
+@[simp] lemma of_lower_to_lower (a : α) : of_lower (to_lower a) = a := rfl
+@[simp] lemma to_lower_inj {a b : α} : to_lower a = to_lower b ↔ a = b := iff.rfl
+@[simp] lemma of_lower_inj {a b : with_lower_topology α} : of_lower a = of_lower b ↔ a = b :=
+iff.rfl
+
+/-- A recursor for `with_lower_topology`. Use as `induction x using with_lower_topology.rec`. -/
+protected def rec {β : with_lower_topology α → Sort*}
+  (h : Π a, β (to_lower a)) : Π a, β a := λ a, h (of_lower a)
+
+instance [nonempty α] : nonempty (with_lower_topology α) := ‹nonempty α›
+instance [inhabited α] : inhabited (with_lower_topology α) := ‹inhabited α›
+
+variables [preorder α]
+
+instance : preorder (with_lower_topology α) := ‹preorder α›
+
+instance : topological_space (with_lower_topology α) := generate_from {s | ∃ a, (Ici a)ᶜ = s}
+
+lemma is_open_preimage_of_lower (S : set α) :
+  is_open (with_lower_topology.of_lower ⁻¹' S) ↔
+    (generate_from {s : set α | ∃ (a : α), (Ici a)ᶜ = s}).is_open S := iff.rfl
+
+lemma is_open_def (T : set (with_lower_topology α)) :
+  is_open T ↔ (generate_from {s : set α | ∃ (a : α), (Ici a)ᶜ = s}).is_open
+    (with_lower_topology.to_lower ⁻¹' T) := iff.rfl
+
+end with_lower_topology
+
+/--
+The lower topology is the topology generated by the complements of the closed intervals to infinity.
+-/
+class lower_topology (α : Type*) [t : topological_space α] [preorder α] : Prop :=
+(topology_eq_lower_topology [] : t = generate_from {s | ∃ a, (Ici a)ᶜ = s})
+
+instance [preorder α] : lower_topology (with_lower_topology α) := ⟨rfl⟩
+
+namespace lower_topology
+
+/-- The complements of the upper closures of finite sets are a collection of lower sets
+which form a basis for the lower topology. -/
+def lower_basis (α : Type*) [preorder α] :=
+{s : set α | ∃ t : set α, t.finite ∧ (upper_closure t : set α)ᶜ = s}
+
+section preorder
+variables [preorder α] [topological_space α] [lower_topology α] {s : set α}
+
+/-- If `α` is equipped with the lower topology, then it is homeomorphic to `with_lower_topology α`.
+-/
+def with_lower_topology_homeomorph : with_lower_topology α ≃ₜ α :=
+{ continuous_to_fun := by { convert continuous_id, apply topology_eq_lower_topology },
+  continuous_inv_fun := by { convert ← continuous_id, apply topology_eq_lower_topology },
+  ..with_lower_topology.of_lower }
+
+lemma is_open_iff_generate_Ici_compl : is_open s ↔ generate_open {s | ∃ a, (Ici a)ᶜ = s} s :=
```

Comment:
```suggestion
lemma is_open_iff_generate_Ici_compl : is_open s ↔ generate_open {t | ∃ a, (Ici a)ᶜ = t} s :=
```
I know Lean doesn't get confused, but a reader could, so better use different names for different objects.

Reply:
Good suggestion, thanks. I easily get confused!